File: my-folder/0206-reverse-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        head=reverse(head,nullptr);
        return head;
    
    }
    ListNode* reverse(ListNode* head, ListNode* prev){
        if(head==NULL) return prev;
        ListNode* temp = head->next;
        head->next = prev;
        prev=head;
        return reverse(temp,prev);
    }
};

File: my-folder/0206-reverse-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode reverse(ListNode head,ListNode prev){
        if(head==null) return prev;
        ListNode temp = head.next;
        head.next = prev;
        prev = head;
        head = temp;
        return reverse(head,prev);
        
    }
    public ListNode reverseList(ListNode head) {
        head  = reverse(head,null);
        return head;
    }
}
